docs/[tech_doc]how_the_consensus_protocol_works.md

```diff
@@ -0,0 +1,94 @@
+# How the consensus protocol works
+
+# Consensus Protocol
+
+There are two protocols in `SEBAK`. One is `ISAAC` protocol for consensus and another is `Transaction Protocol` for sharing transactions.
+
+## `ISAAC` Protocol
+The `ISAAC` is a consensus protocol based on `PBFT`.
+
+## `ISAAC` States
+1. `INIT` - Proposed or received a ballot with transactions.
+
+1. `SIGN` - Checked that the ballot is valid by itself.
+
+1. `ACCEPT` - Checked that the ballot is valid by validators.
+
+1. `ALL_CONFIRM` - Confirmed block with the transactions in the ballot.
+
+## Voting Hole
+
+1. `YES` - Agreed
+
+1. `NO` - Disagreed
+
+1. `EXP` - Expired
+
+## Terms
+
+* TIMEOUT_INIT - The timeout for `INIT` state. The default value is 2 sec.
+
+* TIMEOUT_SIGN - The timeout for `SIGN` state. The default value is 2 sec.
+
+* TIMEOUT_ACCEPT - The timeout for `ACCEPT` state. The default value is 2 sec.
+
+* TIMEOUT_ALLCONFIRM - The timeout for `ALLCONFIRM`. This value is not configurable. It is calculated based on block time and proposed time. Please check [How-to-calculate-timeout-allconfirm](./[tech_doc]how_to_calculate_timeout_allconfirm.md) in detail.
+
+* B(ISAAC State, Voting Hole) - A ballot with ISAAC state and voting hole.
+
+### Examples
+
+* B(`INIT`, `YES`) - A ballot with ISAAC state `INIT` and voting hole `YES`.
+
+* B(`SIGN`, `NO`) - A ballot with ISAAC state `INIT` and voting hole `NO`.
+
+* B(`ACCEPT`, `EXP`) - A ballot with ISAAC state `INIT` and voting hole `EXP`.
+
+## Voting Process
+
+### Network Start
+* At the beginning of the network, the genesis block is saved with block height 1 and round 0.
+* The node start with `INIT` state, height 2 and round 0.
+
+### `INIT`
+1. The timer is set to TIMEOUT_INIT.
+1. The steps to propose transactions are as follows.
+   * If the node is [proposer](./[tech_doc]how_to_select_the_proposer.md) of this round broadcasts a B(`INIT`, `YES`),
+      * The ballot includes valid transactions(only hashes) or empty transaction.
+      * When it broadcasts the ballot, the node goes to the next state.
+   * If the node is not proposer, it waits for and receive the B(`INIT`, `YES`).
+      * When it receives the ballot, the node goes to the next state.
+   * When the timer expires, the node goes to the next state.
+
+### `SIGN`
+1. The timer is reset to TIMEOUT_SIGN
+1. The node checks [the proposed ballot is valid](./[tech_doc]how_to_check_a_ballot_is_valid.md).
+   * If the proposed ballot is valid, the node broadcasts B(`SIGN`, `YES`).
+   * If the proposed ballot is invalid, the node broadcasts B(`SIGN`, `NO`).
+   * If the proposed ballot is empty, the node broadcasts B(`SIGN`, `EXP`).
```

Comment:
@cmcm2222 I checked this with @Charleslee522 
`If the proposed ballot is empty, the node broadcasts B(`SIGN`, `EXP`)` should be changed to
`If the proposed ballot is empty, the node broadcasts B(`SIGN`, `YES`)`

Reply:
OK. I will change.